packages/components/MenuButton/macos/MenuButton.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import AppKit
+
+@objc(MSFMenuButton)
+open class MenuButton: NSPopUpButton {
+
+  public override init(frame buttonFrame: NSRect, pullsDown flag: Bool) {
+
+    super.init(frame: buttonFrame, pullsDown: flag)
+
+    imagePosition = .imageLeading
+    bezelStyle = .recessed
+
+    if #available(OSX 11.0, *) {
+      controlSize = .large
+    }
+
+    guard let dropDownCell = cell as? NSPopUpButtonCell else {
+      preconditionFailure()
+    }
+    dropDownCell.imagePosition = .imageLeading
+    dropDownCell.arrowPosition = .arrowAtBottom
+  }
+
+  @available(*, unavailable)
+  required public init?(coder decoder: NSCoder) {
+    preconditionFailure()
+  }
+
+  @objc public convenience init() {
+    self.init(frame: .zero, pullsDown: true)
+  }
+
+  @objc public var OnItemClick: RCTBubblingEventBlock?
+
+  @objc public var OnSubmenuItemClick: RCTBubblingEventBlock?
+
+  open override var menu: NSMenu? {
+    didSet {
+      updateMenu()
+    }
+  }
+
+  open override var image: NSImage? {
+    get {
+      return menuButtonImage
+    }
+    set {
+      // We must set the image on the dropdown cell rather than the button,
+      // If we set the image on Button itself, no image is displayed.
+      menuButtonImage = newValue
+      updateDropDownCell()
+    }
+  }
+
+  open override var title: String {
+    didSet {
+      updateDropDownCell()
+    }
+  }
+
+  // MARK: - Private Methods
+
+  private func updateMenu() {
+    guard let menu = menu else {
+      return
+    }
+
+    for (index, menuItem) in menu.items.enumerated() {
+      menuItem.target = self
+      menuItem.action = #selector(sendOnItemClickEvent)
+      if menuItem.hasSubmenu, let submenu = menuItem.submenu {
+        //Add actions to one level of submenu items to support the `onSubmenuItemClick` callback
+        for subMenuItem in submenu.items {
+          subMenuItem.tag = index //store the index of the "super" menuItem for lookup in the action
+          subMenuItem.target = self
+          subMenuItem.action = #selector(sendOnSubItemClickEvent)
+        }
+      }
+    }
+
+    // Insert an initial empty item into index 0, since index 0 is never displayed
+    // We must do this after we assign the tags to the submenuItems to preserve index order
```

Comment:
nit: Add a period at the end of both lines.  Comments can either be a phrase, or complete sentence(s).  Multiline comments containing multiple sentences should be proper paragraphs.